android/app/src/main/java/io/mosip/registration_client/api_services/RegistrationApi.java

```diff
@@ -0,0 +1,94 @@
+package io.mosip.registration_client.api_services;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.mosip.registration.clientmanager.dto.registration.RegistrationDto;
+import io.mosip.registration.clientmanager.dto.uispec.FieldSpecDto;
+import io.mosip.registration.clientmanager.service.TemplateService;
+import io.mosip.registration.clientmanager.spi.RegistrationService;
+import io.mosip.registration.clientmanager.util.UserInterfaceHelperService;
+import io.mosip.registration.packetmanager.util.JsonUtils;
+import io.mosip.registration_client.model.RegistrationDataPigeon;
+
+@Singleton
+public class RegistrationApi implements RegistrationDataPigeon.RegistrationDataApi {
+    private final RegistrationService registrationService;
+    RegistrationDto registrationDto;
+    TemplateService templateService;
+
+    @Inject
+    public RegistrationApi(RegistrationService registrationService, TemplateService templateService) {
+        this.registrationService = registrationService;
+        this.templateService = templateService;
+    }
+
+    @Override
+    public void startRegistration(@NonNull List<String> languages, @NonNull RegistrationDataPigeon.Result<String> result) {
+        String response = "";
+        try {
+            this.registrationDto = registrationService.startRegistration(languages);
+            result.success(response);
```

Comment:
are we returning "" everytime if start reg is success, then line 40 may not be needed since the same is performed in 46 as well